docs/monitoring.md

```diff
@@ -1,32 +1,83 @@
 # Monitoring
 
-## Monitoring container resource usage
+## Prometheus
 
-When run in docker, a few additional containers are started up:
+To monitor RPC return values over time, a Prometheus data exporter can be connected
+to any Bitcoin Tank and configured to scrape any available RPC results.
 
-* CAdvisor (container Monitoring)
-* Prometheus (log scraper)
-* Grafana (graphing/dashboard tool)
+The `bitcoin-exporter` image is defined in `resources/images/exporter` and
+maintained in the BitcoinDevProject dockerhub organization. To add the exporter
+in the Tank pod with Bitcoin Core add the `"exporter"` key to the node in the graphml file:
 
-## CAdvisor
+```xml
+    <node id="0">
+        <data key="version">27.0</data>
+        <data key="exporter">true</data>
+    </node>
+```
 
-CAdvisor needs no additional setup, and can be accessed from the docker host at
-localhost:8080
+The default metrics are defined in the `bitcoin-exporter` image:
+- Block count
+- Number of inbound peers
+- Number of outbound peers
+- Mempool size (# of TXs)
 
-## Prometheus
+Metrics can be configured by setting a `"metrics"` key to the node in the graphml file.
+The metrics value is a space-separated list of labels, RPC commands with arguments, and
+JSON keys to resolve the desired data:
+
+```
+label=method(arguments)[JSON result key][...]
+```
+
+For example, the default metrics listed above are defined as:
+
+```xml
+    <node id="0">
+        <data key="version">27.0</data>
+        <data key="exporter">true</data>
+        <data key="metrics">blocks=getblockcount() inbounds=getnetworkinfo()["connections_in"] outbounds=getnetworkinfo()["connections_in"] mempool_size=getmempoolinfo()["size"]</data>
```

Comment:
Let's add a `metrics` key to both default graph and generated graphs from `warcli graph create` so that users can easily add this key without getting `missing key` (at the graph level) errors.

Reply:
```log
warnet-rpc  | 2024-08-06 09:11:13 | ERROR   | server   | Error bring up warnet: Bad GraphML data: no key metrics
2024-08-06 09:11:13 | ERROR   | warnet.server | jsonrpc error
2024-08-06 09:11:13 | ERROR   | warnet.server | 
Traceback (most recent call last):
warnet-rpc  |   File "/usr/local/lib/python3.12/site-packages/networkx/readwrite/graphml.py", line 966, in decode_data_elements
    data_name = graphml_keys[key]["name"]
warnet-rpc  |                 ~~~~~~~~~~~~^^^^^
KeyError: 'metrics'
```

Reply:
cool done